packages/react/src/auto/polaris/inputs/PolarisTimePicker.tsx

```diff
@@ -0,0 +1,205 @@
+    props.fieldProps.onChange(getDateFromDateTimeObject({ ...getDateTimeObjectFromDate(props.fieldProps.value ?? new Date()), hour }));
+  const setMinSelected = (minute: string) =>
+    props.fieldProps.onChange(getDateFromDateTimeObject({ ...getDateTimeObjectFromDate(props.fieldProps.value ?? new Date()), minute }));
+  const setAmpmSelected = (ampm: string) =>
+    props.fieldProps.onChange(getDateFromDateTimeObject({ ...getDateTimeObjectFromDate(props.fieldProps.value ?? new Date()), ampm }));
+
+  const onTimeStringChange = (value: string) => {
+    setTimeString(value);
+    let trimmedValue = value.trim();
+    let hours = parseInt(trimmedValue.split(":")[0], 10);
+    const suffix = trimmedValue.split(":")[1];
+
+    if (hours === 0) {
+      trimmedValue = "12:" + suffix;
+    }
+
+    const result = timeFormatRegex.test(trimmedValue);
+
+    if (!result) {
+      setTimeParseError(true);
+      return;
+    }
+
+    setTimeParseError(false);
+
+    const minutes = parseInt(suffix, 10);
+    const newAmpm = trimmedValue.slice(-2).toUpperCase();
+    const hoursStr = hours.toString();
+    const paddedMins = minutes.toString().padStart(2, "0");
+    setHourSelected(hoursStr);
+    setMinSelected(paddedMins);
+    setAmpmSelected(newAmpm);
+
+    const parsedDate = new Date();
+    if (newAmpm === "PM" && hours < 12) {
+      hours += 12;
+    } else if (newAmpm === "AM" && hours === 12) {
+      hours = 0;
+    }
+
+    parsedDate.setHours(hours, minutes, 0, 0);
+    const date = props.localTime ? new Date(props.localTime) : new Date();
+
+    copyTime(date, parsedDate);
+
+    props.onChange?.(zonedTimeToUtc(date, props.localTz ?? "UTC"));
+    props.fieldProps.onChange(zonedTimeToUtc(date, props.localTz ?? "UTC"));
+
+    setTimeString(`${parseInt(hoursStr, 10) === 0 ? "12" : hoursStr}:${paddedMins} ${newAmpm}`);
+  };
+
+  useEffect(() => {
+    if (!props.fieldProps.value) return;
+    setTimeString(getTimeString(getDateTimeObjectFromDate(props.fieldProps.value)));
+  }, [props.fieldProps.value]);
+
+  return (
+    <>
+      <Popover
+        active={timePopoverActive}
+        onClose={() => setTimePopoverActive(false)}
+        activator={
+          <TextField
+            prefix={<Icon source={ClockIcon} />}
+            id={props.id ? `${props.id}-time` : undefined}
+            data-testid={props.id ? `${props.id}-time` : undefined}
+            label={props.timeLabel ?? "Time"}
+            autoComplete="off"
+            value={timeString}
+            onChange={onTimeStringChange}
+            onFocus={() => setTimePopoverActive(true)}
+            error={timeParseError && `Invalid time format`}
+            {...props.timePickerProps}
+          />
+        }
+      >
+        {!props.hideTimePopover && (
+          <div style={{ overflow: "hidden", padding: "15px 8px" }}>
+            <div style={{ display: "flex" }}>
+              <Scrollable style={{ overflowY: "scroll", height: "250px" }}>
```

Comment:
This is a Scrollable while the others are all div

Consider making a component for a single scroller and then do something like this
```jsx
[hoursProps, minutesProps, amPmProps].map((props)=> <ScrollableTimeOptions {...props}/>)
```
That way, we can avoid these `changed one but should have changed all of them` moments